src/Diff.elm

```diff
@@ -0,0 +1,319 @@
+module Diff exposing (Change(..), diff, diffLines)
+
+-- NOTE: This is copy/pasted from https://github.com/jinjor/elm-diff
+-- It's inlined to avoid having to call elm-package install on the end user's
+-- system - the approach this library took prior to
+-- commit 19047f01d460739bfe7f16466bc60b41430a8f09 - because it assumes
+-- the end user has the correct elm-package on their PATH, which is not a
+-- safe assumption.
+--
+-- License:
+{-
+   Copyright (c) 2016, Yosuke Torii
+   All rights reserved.
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are met:
+
+   * Redistributions of source code must retain the above copyright notice, this
+     list of conditions and the following disclaimer.
+
+   * Redistributions in binary form must reproduce the above copyright notice,
+     this list of conditions and the following disclaimer in the documentation
+     and/or other materials provided with the distribution.
+
+   * Neither the name of elm-diff nor the names of its
+     contributors may be used to endorse or promote products derived from
+     this software without specific prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+   FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+   SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+   CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+   OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-}
+
+import Array.Hamt as Array exposing (Array)
```

Comment:
Does it make sense to use core's array? On principle, I agree with using `Array.Hamt` _but_ it adds an extra dependency, and requires extra code to ensure that dependency is there. I don't see unsafe usage (it's mostly just access by index and `Array.fromList`, rather than slicing and comparing), so I think that's feasible.